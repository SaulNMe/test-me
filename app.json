{
  "expo": {
    "name": "Qualificame",
    "slug": "qualificame-native",
    "privacy": "public",
    "sdkVersion": "33.0.0",
    "platforms": [
      "ios",
      "android"
    ],
    "version": "1.0.0",
    "orientation": "portrait",
    "icon": "./assets/icon.png",
    "splash": {
      "image": "./assets/splashCustom.png",
      "resizeMode": "cover",
      "backgroundColor": "#ffffff",

    },
    "updates": {
      "fallbackToCacheTimeout": 0
    },
    "assetBundlePatterns": [
      "**/*"
    ],
    "ios": {
      "supportsTablet": true
    }
  }
}
