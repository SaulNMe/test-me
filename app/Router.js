import React, { Component } from 'react';
import { Platform, Animated, Easing } from 'react-native';
import { createAppContainer, createStackNavigator, createBottomTabNavigator, createSwitchNavigator } from 'react-navigation';
import WelcomeScreen from 'qualificame-native/app/screens/WelcomeScreen';
import LoginScreen from 'qualificame-native/app/screens/LoginScreen';
import SignUpScreen from 'qualificame-native/app/screens/SignUpScreen';
import CreateAccountScreen from 'qualificame-native/app/screens/CreateAccountScreen';
import SelectPlanScreen from 'qualificame-native/app/screens/SelectPlanScreen';
import PaymentScreen from 'qualificame-native/app/screens/PaymentScreen';
import KiosksScreen from 'qualificame-native/app/screens/KiosksScreen';
import CreateKioskScreen from 'qualificame-native/app/screens/CreateKioskScreen';
import KioskDetailScreen from 'qualificame-native/app/screens/KioskDetailScreen';
import KioskConfigurationScreen from 'qualificame-native/app/screens/KioskConfigurationScreen';
import KioskContactDataScreen from 'qualificame-native/app/screens/KioskContactDataScreen';
import KioskOptionsScreen from 'qualificame-native/app/screens/KioskOptionsScreen';
import ReportsScreen from 'qualificame-native/app/screens/ReportsScreen';
import CreateReportScreen from 'qualificame-native/app/screens/CreateReportScreen';
import SummaryReportScreen from 'qualificame-native/app/screens/SummaryReportScreen';
import ReportDetailScreen from 'qualificame-native/app/screens/ReportDetailScreen';
import MailBoxScreen from 'qualificame-native/app/screens/MailBoxScreen';
import MyAccountScreen from 'qualificame-native/app/screens/MyAccountScreen';
import ProPlanScreen from 'qualificame-native/app/screens/ProPlanScreen';
import EnterprisePlanScreen from 'qualificame-native/app/screens/EnterprisePlanScreen';
import ChangePasswordScreen from 'qualificame-native/app/screens/ChangePasswordScreen';
import FaqScreen from 'qualificame-native/app/screens/FaqScreen';
import AlertsScreen from 'qualificame-native/app/screens/AlertsScreen';
import BottomBar from 'qualificame-native/app/components/BottomBar';
import LoadingScreen from 'qualificame-native/app/screens/LoadingScreen';
import SendEmailScreen from 'qualificame-native/app/screens/SendEmailScreen';
import VerificationCodeScreen from 'qualificame-native/app/screens/VerificationCodeScreen';
import ConfirmPasswordScreen from 'qualificame-native/app/screens/ConfirmPasswordScreen';
import AuthLoadingScreen from 'qualificame-native/app/screens/AuthLoadingScreen';
import ScannerScreen from 'qualificame-native/app/screens/ScannerScreen';

import GetStateForAction from 'qualificame-native/app/utils/GetStateForAction';
import TransitionConfiguration from 'qualificame-native/app/utils/TransitionConfiguration';

const RecoverPasswordStack = createStackNavigator({
	SendEmailScreen,
	VerificationCodeScreen,
	ConfirmPasswordScreen,
	LoadingScreen: { 
		screen: LoadingScreen,
		navigationOptions: null
	},
}, {
	transitionConfig: TransitionConfiguration,
	navigationOptions: {
		header: null
	}
})

const AuthModalStackNavigator = createStackNavigator({
	WelcomeScreen,
	LoginScreen,
	RecoverPasswordStack,
	SignUpScreen,
	CreateAccountScreen,
	SelectPlanScreen,
	PaymentScreen,
	LoadingScreen: { 
		screen: LoadingScreen,
		navigationOptions: null
	},
}, {
	transitionConfig: TransitionConfiguration
});

const KioskConfigurationStack = createStackNavigator({
	KioskConfigurationScreen,
	KioskOptionsScreen,
	KioskContactDataScreen
});

const CreateReportStack = createStackNavigator({
	CreateReportScreen,
	SummaryReportScreen,
	LoadingScreen,
}, {
	transitionConfig: TransitionConfiguration,
	navigationOptions: {
		header: null
	}
});

GetStateForAction(CreateReportStack);

const BottomBarNavigator = createBottomTabNavigator({
	KiosksScreen,
	ReportsScreen,
	MailBoxScreen,
	MyAccountScreen
}, {
	initialRouteName: 'KiosksScreen',
	tabBarComponent: BottomBar
});

const CameraModalNavigator = createStackNavigator({
	ScannerScreen
}, {
	mode: 'modal',
	navigationOptions: {
		header: null,
	}
})

const MainStackNavigator = createStackNavigator({
	BottomBarNavigator: {
		screen: BottomBarNavigator,
		navigationOptions: {
			header: null,
		}
	},
	KioskDetailScreen,
	CameraModalNavigator,
	ReportDetailScreen,
	ProPlanScreen,
	EnterprisePlanScreen,
	ChangePasswordScreen,
	FaqScreen
}, {
	initialRouteName: 'BottomBarNavigator',
});

const ModalStack = createStackNavigator({
	MainStackNavigator,
	AlertsScreen,
	CreateKioskScreen,
	KioskConfigurationStack,
	CreateReportStack
}, {
	mode:'modal',
	headerMode: 'none'
});

const MainNavigation = createSwitchNavigator({
	AuthLoading: AuthLoadingScreen,
	Auth: AuthModalStackNavigator,
	App: ModalStack,
}, {
	initialRouteName: 'AuthLoading'
});

export default createAppContainer(MainNavigation);




 




