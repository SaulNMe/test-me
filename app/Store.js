import { createStore, applyMiddleware, compose } from 'redux';
import { persistStore, persistReducer } from 'redux-persist';
import autoMergeLevel1 from 'redux-persist/lib/stateReconciler/autoMergeLevel1';
import storage from 'redux-persist/lib/storage';
import thunk from 'redux-thunk'
import rootReducer from 'qualificame-native/app/reducers'
import Reactotron from 'qualificame-native/config/ReactotronConfig'

const persistConfig = {
	key: 'root',
	storage: storage,
	stateReconciler: autoMergeLevel1,
	whitelist: ['Auth']
};

const pReducer = persistReducer(persistConfig, rootReducer);

let store = createStore(
	pReducer,
	compose(
		applyMiddleware(thunk),
		Reactotron.createEnhancer()
	)
);

export default store;
export const persistor = persistStore(store);