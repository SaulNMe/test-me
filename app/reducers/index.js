import Events, * as fromEvents from 'qualificame-native/app/reducers/EventsReducer.js';
import Auth, * as fromAuth from 'qualificame-native/app/reducers/AuthReducer.js';
import TokenRefresh, * as fromTokenRefresh from 'qualificame-native/app/reducers/TokenRefreshReducer.js'
import Kiosks, * as fromKiosks from 'qualificame-native/app/reducers/KiosksReducer.js';
import Reports, * as fromReports from 'qualificame-native/app/reducers/ReportsReducer.js';
import Choices, * as fromChoices from 'qualificame-native/app/reducers/ChoicesReducer.js';
import Alerts, * as fromAlerts from 'qualificame-native/app/reducers/AlertsReducer.js';

import { combineReducers } from 'redux'

const rootReducer = combineReducers({
    Auth,
    TokenRefresh,
    Kiosks,
    Reports,
    Events,
    Choices,
    Alerts,
});

export default rootReducer;

//Auth
export const getIsLoggedIn = state =>
	fromAuth.getIsLoggedIn(state.Auth);

export const getIsLoading = state =>
	fromAuth.getIsLoading(state.Auth);

export const getAuthData = state =>
	fromAuth.getAuthData(state.Auth);

export const getUserData = state =>
	fromAuth.getUserData(state.Auth);

//Token Refresh
export const getIsRefreshingToken = state =>
	fromTokenRefresh.getIsRefreshingToken(state.TokenRefresh);

export const getRefreshCall = state =>
	fromTokenRefresh.getRefreshCall(state.TokenRefresh);

//Kiosks
export const getKiosks = state =>
	fromKiosks.getKiosks(state.Kiosks);

export const getKioskById = (state, kioskId) =>
	fromKiosks.getKioskById(state.Kiosks, kioskId)

export const getIsLoadingKiosks = state =>
	fromKiosks.getIsLoadingKiosks(state.Kiosks);

export const getErrorKiosks = state =>
	fromKiosks.getErrorKiosks(state.Kiosks);

//Reports

export const getReports = state =>
	fromReports.getReports(state.Reports);

export const getIsLoadingReports = state =>
	fromReports.getIsLoadingReports(state.Reports);
export const getErrorReports = state =>
	fromReports.getErrorReports(state.Reports);

// Events
export const getEvents = state =>
	fromEvents.getEvents(state.Events);

export const getEventsGroupedByDay = (state, ids) =>
	fromEvents.getEventsGroupedByDay(state.Events, ids);

export const getSelectedMonth = state => 
	fromEvents.getSelectedMonth(state.Events);

export const getEventsSelectedDateRange = state => 
	fromEvents.getEventsSelectedDateRange(state.Events);

// Choices
export const getChoices = state =>
	fromChoices.getChoiceById(state.Choices);

// Alerts 
export const getAlerts = state =>
	fromAlerts.getAlerts(state.Alerts);

export const unReadAlerts = state =>
	fromAlerts.unReadAlerts(state.Alerts);