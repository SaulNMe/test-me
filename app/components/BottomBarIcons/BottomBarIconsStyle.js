import { StyleSheet } from 'react-native';
import { ApplicationStyles, Fonts, Colors, Metrics } from 'qualificame-native/app/styles';

export default StyleSheet.create({
	...ApplicationStyles.screen,
	...ApplicationStyles.flexBox,
	...ApplicationStyles.margins,
	pillStyle: {
		borderRadius: 100,
		width: 100,
	}
});
