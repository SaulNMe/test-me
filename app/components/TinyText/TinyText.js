import React, { PureComponent } from 'react';
import { Text } from 'react-native';
import PropTypes from 'prop-types';
import { Fonts, Colors } from 'qualificame-native/app/styles';
import styles from './TinyTextStyle';

export default class TinyText extends PureComponent {
	render() {
		const textColor = {color: Colors[this.props.color]};
		const weight = {fontWeight: Fonts.weight[this.props.weight]};
		const align = {textAlign: this.props.align};

		return (
			<React.Fragment>
				<Text style={[styles.text, textColor, weight, align]}>{this.props.text}</Text>
			</React.Fragment>
		);
	}
}

	TinyText.propTypes = {
		text: PropTypes.string,
		color: PropTypes.string,
		weight: PropTypes.string,
		align: PropTypes.string,
	}

	TinyText.defaultProps = {
		text: "",
		color: "dark",
		weight: "regular",
		align: "left"
	}
