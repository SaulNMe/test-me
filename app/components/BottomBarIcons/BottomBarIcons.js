import React, { PureComponent } from 'react';
import { Text, View, Animated, Platform, TouchableOpacity } from 'react-native';
import PropTypes from 'prop-types';
import { Colors } from 'qualificame-native/app/styles';

import styles from './BottomBarIconsStyle';
import { Feather } from '@expo/vector-icons';
import TinyText from 'qualificame-native/app/components/TinyText';

import NavigationService from 'qualificame-native/app/services/NavigationService'; 
export default class BottomBarIcons extends PureComponent {
	_width = new Animated.Value(41);
	_widthText = new Animated.Value(50);
	_heightText = new Animated.Value(Platform.OS === 'ios' ? 15 : 16.3);
	state = {
		isActualScreen: this.props.isActualScreen,
		routeName: this.props.routeName,
	}

	componentDidMount() {
		if(this.props.isActualScreen){
			setTimeout(() => {
				this.handleWidth();
			},400)
		}		
	}
	
	componentDidUpdate(prevProps, prevState) {
		setTimeout(() => {
			if(this.props.isActualScreen){
				this.handleWidth();
			} else {
				this.resetWidth();
			}
		},400)
	}

	switchValue = () => {
		switch(this.state.routeName){
			case 'KiosksScreen':
				this._widthText = Platform.OS === 'ios' ? 46 : 44 
				return this._widthText + 50;
			case 'ReportsScreen':
				this._widthText = Platform.OS === 'ios' ? 54 : 50
				return this._widthText + 50;
			case 'MailBoxScreen':
				this._widthText = Platform.OS === 'ios' ? 37 : 34
				return this._widthText + 50;
			case 'MyAccountScreen':
				this._widthText = Platform.OS === 'ios' ? 31.5 : 31
				return this._widthText + 50;
		}
	}
	
	handleWidth = () => {
		Animated.timing(
			this._width, {
				toValue: this.switchValue(),
				duration: 200
			}
		).start();
	}

	resetWidth = () => {
		Animated.timing(
			this._width, {
				toValue: 41,
				duration: 200
			}
		).start();		
	}
	
	handleOnPress = () => {
		if(!this.props.isActualScreen){
 			this.props.onPress(this.props.routeName);
 			NavigationService.navigate(this.props.routeName);
		}
	}

	render() {
		let color = this.props.isActualScreen ? Colors[this.props.color] : '#364350';
		let stylesActualScreen = this.props.isActualScreen ? [
					styles.pillStyle,
					styles.row,
					styles.alignItemsCenter,
					styles.smallVerticalPadding,
					styles.smallHorizontalPadding,
					{ overflow: 'hidden', backgroundColor: Colors[this.props.bgColor]}] : [];

		return (
			<TouchableOpacity 
					onPress={() => this.handleOnPress()}
					style={styles.centerObjects}
			>
				<Animated.View style={[stylesActualScreen, (this.props.isActualScreen && {width: this._width})]}>
					<Feather 
						name={this.props.iconName} 
						size={25}
						color={color}
					/>
					{ this.props.isActualScreen && 
						<Animated.View style={[styles.smallLeftMargin, {width: this._widthText, height: this._heightText, overflow: 'hidden'}]}>
							<TinyText 
								text={this.props.name}
								color={this.props.color}
								weight="medium"
							/>
						</Animated.View>
					}
				</Animated.View>
		     </TouchableOpacity>

		);
	}
}

BottomBarIcons.propTypes = {
	routeName: PropTypes.string,
	isActualScreen: PropTypes.bool,
	name: PropTypes.string,
	color: PropTypes.string,
	iconName: PropTypes.string,
	bgColor: PropTypes.string,
}

BottomBarIcons.defaultProps = {
	routeName: '',
	isActualScreen: false,
	name: 'Kioskos' ,
	color: 'green' ,
	iconName: 'home' ,
	bgColor: 'lightGreen' ,
}
