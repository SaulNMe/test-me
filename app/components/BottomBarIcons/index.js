import BottomBarIcons from './BottomBarIcons';

export default BottomBarIcons;
