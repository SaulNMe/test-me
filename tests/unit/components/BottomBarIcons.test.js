import React from 'react';
import { shallow } from 'enzyme';
import { Text } from 'react-native';

import BottomBarIcons from 'qualificame-native/app/components/BottomBarIcons';
import { Feather } from '@expo/vector-icons';
import TinyText from 'qualificame-native/app/components/TinyText';
import { Colors } from 'qualificame-native/app/styles';


function createTestProps (props) {
	return {
		// common props
		routeName: 'KiosksScreen',
		iconName: 'home',
		isActualScreen: true,
		// allow to override common props
		...props,
	}
}

describe('rendering', () => {
	let wrapper;
	const createWrapper = props => shallow(<BottomBarIcons {...props}/>);
	beforeEach(() => {
		const props = createTestProps()
		wrapper = createWrapper(props);
	});

	it('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('displays the correct text ', () => {
		expect(wrapper.find(TinyText).prop('text')).toEqual('Kioskos');
	});	

	it('displays the correct feather icon', () => {
		expect( wrapper.find(Feather).prop('name')).toEqual('home');
	});

	describe('with no props', ()=>{
		beforeEach(() => {
			wrapper = createWrapper({ routeName: 'KiosksScreen' });
		});

		it('displays no text', () => {
			expect(wrapper.find(TinyText)).toEqual({});
		});	

		it('displays the correct feather icon', () => {
			expect( wrapper.find(Feather).prop('name')).toEqual('home');
		});
	});
});
