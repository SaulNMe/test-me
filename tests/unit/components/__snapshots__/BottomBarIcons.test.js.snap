// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders as expected 1`] = `
<Component
  style={
    Array [
      Object {
        "borderRadius": 100,
        "width": 100,
      },
      Object {
        "flexDirection": "row",
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {
        "paddingVertical": 8,
      },
      Object {
        "paddingHorizontal": 8,
      },
      Object {
        "backgroundColor": "#DAF0E2",
      },
    ]
  }
>
  <Component
    style={
      Object {
        "marginRight": 8,
      }
    }
  >
    <Icon
      allowFontScaling={false}
      color="#14A286"
      name="home"
      size={25}
    />
  </Component>
  <TinyText
    align=""
    color="green"
    text="Kioskos"
    weight="medium"
  />
</Component>
`;
